node/database-cleaner/src/appwrite.js

```diff
@@ -0,0 +1,86 @@
+import { Client, Databases, Query } from 'node-appwrite';
+import { getExpiryDate } from './utils.js'
+
+class AppwriteService {
+  constructor() {
+    const client = new Client();
+    client
+      .setEndpoint(
+        process.env.APPWRITE_ENDPOINT ?? 'https://cloud.appwrite.io/v1'
+      )
+      .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
+      .setKey(process.env.APPWRITE_API_KEY);
+
+    this.databases = new Databases(client);
+  }
+
+  /**
+   * @param {string} databaseId
+   * @returns {Promise<Models.CollectionList>}
+   */
+  async listAllCollections(databaseId) {
+    const totalCollections = [];
+    const queries = [Query.orderAsc('$createdAt'), Query.limit(25)];
+    let done = false;
+
+    do {
+      const collections = await this.databases.listCollections(
+        databaseId,
+        queries
+      );
+
+      const lastCollectionId =
+        collections.collections[collections.collections.length - 1].$id;
+
+      const cursorPos = queries.findIndex((x) => x.includes('cursorAfter'));
+
+      if (cursorPos !== -1) {
+        queries.splice(cursorPos, 1);
+      }
+
+      queries.push(Query.cursorAfter(lastCollectionId));
+
+      totalCollections.push(...collections.collections);
+
+      if (totalCollections.length === collections.total) {
+        done = true;
+      }
+    } while (!done)
```

Comment:
Can you move the while condition inline?

Reply:
I don't think it can be moved inline, the variables within the do part of the while loop are not accessible in the while condition. I can do a while look and break if a condition is met, but I think I have to have an outside variable tracking if the loop is finished.